Makefile

```diff
@@ -1,162 +1,186 @@
+IS_WSL  := $(if $(findstring Microsoft,$(shell uname -a)),WSL,)
+IS_OSX  := $(filter Darwin,$(shell uname -a))
+IS_LINUX:= $(if $(or $(IS_WSL),$(IS_OSX)),,$(filter Linux,$(shell uname -a)))
+endif
+IS_WIN  := $(strip $(if $(or $(IS_LINUX),$(IS_OSX),$(IS_WSL)),,$(OS)))
+
+$(if $(IS_WIN),$(error Windows is not supported in all recipes. Use WSL instead. Follow instructions in README.txt),)
+
+# Makefile's shell
+SHELL := /bin/bash
+
+# Host machine IP
+export MACHINE_IP = $(shell source $(realpath $(CURDIR)/scripts/portable.sh) && get_this_ip)
+
+include repo.config
+
+SERVICES = $(sort $(dir $(wildcard services/*/.)))
+# TARGETS --------------------------------------------------
+.DEFAULT_GOAL := help
+
+certificates/domain.crt: certificates/domain.key
+certificates/domain.key:
+	# domain key/crt files must be located in $< and certificates/domain.crt to be used
+	@echo -n "No $< certificate detected, do you wish to create self-signed certificates? [y/N] " && read ans && [ $${ans:-N} = y ]; \
+	$(MAKE) -C certificates create-certificates; \
+	$(MAKE) -C certificates install-root-certificate;
+
+.PHONY: .create-secrets
+.create-secrets:
+	$(MAKE) -C certificates deploy
+
+.PHONY: up-local
+up-local: .install-fqdn certificates/domain.crt certificates/domain.key .create-secrets ## deploy osparc ops stacks and simcore
+	bash scripts/local-deploy.sh
+	@$(MAKE) info-local
+
+.PHONY: up-devel
+up-devel: .install-fqdn certificates/domain.crt certificates/domain.key .create-secrets ## deploy osparc ops stacks and simcore
+	bash scripts/local-deploy.sh --devel_mode=1
+	@$(MAKE) info-local
+
+.PHONY: up-aws
+up-aws:
+	bash scripts/aws-deploy.sh
+
+
+.PHONY: down
+down:
+	@for service in $(SERVICES); do \
+		pushd $$service; $(MAKE) down; popd; \
+	done
+
+.PHONY: leave
+leave: ## leaves the swarm
+	docker swarm leave -f
+
+.PHONY: .install-fqdn
+.install-fqdn:
+	@$(if $(IS_WSL), \
+		if ! grep -Fq "$(MACHINE_IP) $(MACHINE_FQDN)" /c/Windows/System32/drivers/etc/hosts; then \
+		echo -n "Do you wish to install the following host on the Windows host ? [y/N]" && read ans && [ $${ans:-N} = y ] &&  \
+		( echo "please run the following in a PWshell with Admin rights:" && \
+		echo "Add-Content c:\Windows\System32\drivers\etc\hosts \"\`r\`$(MACHINE_IP) $(MACHINE_FQDN)\`r\`$(MACHINE_IP) $(MONITORING_DOMAIN)\`r\`$(MACHINE_IP) $(REGISTRY_DOMAIN)\`r\`$(MACHINE_IP) $(PORTAINER_DOMAIN)\`r\`$(MACHINE_IP) $(STORAGE_DOMAIN)\"" && \
+		echo "OR please run the following in a CMD with Admin rights (note that wildcards are not accepted):" && \
+		echo "echo $(MACHINE_IP) $(MACHINE_FQDN) >> c:\Windows\System32\drivers\etc\hosts && echo $(MACHINE_IP) $(MONITORING_DOMAIN) >> c:\Windows\System32\drivers\etc\hosts && echo $(MACHINE_IP) $(PORTAINER_DOMAIN) >> c:\Windows\System32\drivers\etc\hosts && echo $(MACHINE_IP) $(REGISTRY_DOMAIN) >> c:\Windows\System32\drivers\etc\hosts && echo $(MACHINE_IP) $(STORAGE_DOMAIN) >> c:\Windows\System32\drivers\etc\hosts") \
+		|| [ 42 ]; \
```

Comment:
|| [ 42 ] ? what is this the universal number?

Reply:
It has to be something that need to be true and I in that case I like to put 42, because why not

Reply:
Please consider readability for outcomes. They might not understand not be familiar with bash so using some form of more conventional "true" value might be easier for the rest.

Reply:
Well that's the only fantasy that I allows to myself, and I would like to keep it like this, but I will add a comment to make it easier.